### DISCUSSION TITLE
Rustic does not recognise the file path with spaces and a trailing back slash on Windows

### DISCUSSION THREAD

**Discussion by ghost** [rustic-rs/rustic#1041] [Things are not working as expected - is this a bug?] [answered]
rustic version 0.5.4.

On PowerShell 5.1:

```
D:\> rustic -r "D:\a repo" repoinfo
using no config file (.\rustic.toml doesn't exist)
D:\> rustic -r "D:\a repo\" repoinfo
error: 'rustic.exe' requires a subcommand but one was not provided
  [subcommands: backup, cat, config, completions, check, copy, diff, dump, forget, init, key, list, ls, merge, snapshots, show-config, self-update, prune, restore, repair, repoinfo, tag, help]

Usage: rustic.exe [OPTIONS] <COMMAND>

For more information, try '--help'.
D:\> rustic repoinfo -r "D:\a repo"
using no config file (.\rustic.toml doesn't exist)
enter repository password:
D:\> rustic repoinfo -r "D:\a repo\"
using no config file (.\rustic.toml doesn't exist)
Error: The filename, directory name, or volume label syntax is incorrect. (os error 123)
D:\> rustic -r 'D:\a repo' repoinfo
using no config file (.\rustic.toml doesn't exist)
enter repository password:
D:\> rustic -r 'D:\a repo\' repoinfo
error: 'rustic.exe' requires a subcommand but one was not provided
  [subcommands: backup, cat, config, completions, check, copy, diff, dump, forget, init, key, list, ls, merge, snapshots, show-config, self-update, prune, restore, repair, repoinfo, tag, help]

Usage: rustic.exe [OPTIONS] <COMMAND>

For more information, try '--help'.
D:\> rustic repoinfo -r 'D:\a repo'
using no config file (.\rustic.toml doesn't exist)
enter repository password:
D:\> rustic repoinfo -r 'D:\a repo\'
using no config file (.\rustic.toml doesn't exist)
Error: The filename, directory name, or volume label syntax is incorrect. (os error 123)
```

On PowerShell 7.4.1 all commands above work fine.

**Comment by aawsome** [comment]
Thanks for opening this issue.

In fact, this seems to be a powershell problem and not a rustic one. rustic relies on the caller to put correct commands in the `argv` table - which obviously isn't the case here (as you can see: swapping your parameters leads to a different behavior, which in this case doesn't make sense as these parameters are not positional).

I don't use powershell, but according to https://www.howtogeek.com/694949/how-to-escape-spaces-in-file-paths-on-the-windows-command-line/ your escaping is wrong.
Can you try to use `rustic -r "D:\a repo\" repoinfo`?

**Comment by ghost** [comment]
> Can you try to use `rustic -r "D:\a repo\" repoinfo`?

Hi! I've updated my comment above. Please check it there.

**Comment by aawsome** [comment, marked as answer]
@Refutable4890 If all this works with Powershell 7.4, but not with Powershell 5.1, this is something we cannot solve in rustic. Please refer to the Powershell 5.1 docu how to pass paths with spaces and trailing back slashes to commands.

I can however suggest using a config file for rustic, see https://github.com/rustic-rs/rustic/blob/main/config/README.md.
